packages/playwright-core/src/server/android/android.ts

```diff
@@ -336,42 +341,61 @@ export class AndroidDevice extends SdkObject {
   }
 
   private async _refreshWebViews() {
-    const sockets = (await this._backend.runCommand(`shell:cat /proc/net/unix | grep webview_devtools_remote`)).toString().split('\n');
+    const sockets = (await this._backend.runCommand(`shell:cat /proc/net/unix | grep devtools_remote`)).toString().split('\n');
     if (this._isClosed)
       return;
 
     const newPids = new Set<number>();
+    const socketNames = new Set<string>();
     for (const line of sockets) {
-      const match = line.match(/[^@]+@webview_devtools_remote_(\d+)/);
-      if (!match)
-        continue;
-      const pid = +match[1];
-      newPids.add(pid);
-    }
-    for (const pid of newPids) {
-      if (this._webViews.has(pid))
+      const match = line.match(/[^@]+@.*?devtools_remote_?(\d*)/);
+      const macthSocketName = line.match(/[^@]+@(.*?_devtools_remote_?.*)/);
+
+      if (!macthSocketName)
         continue;
 
-      const procs = (await this._backend.runCommand(`shell:ps -A | grep ${pid}`)).toString().split('\n');
-      if (this._isClosed)
-        return;
-      let pkg = '';
-      for (const proc of procs) {
-        const match = proc.match(/[^\s]+\s+(\d+).*$/);
-        if (!match)
+      const socketName = macthSocketName[1];
+      socketNames.add(socketName);
+
+      if (!match) {
+        if (this._webViews.has(socketName))
           continue;
-        const p = match[1];
-        if (+p !== pid)
+        const webView = { pid: -1, pkg: '', socketName };
```

Comment:
When does this happen? We should always be able to get `pid` and resolve `pkg`, regardless of whether we are selecting by pkg or socketName?